libfwupdplugin/fu-plugin.c

```diff
@@ -1650,6 +1650,8 @@ void
 fu_plugin_add_udev_subsystem (FuPlugin *self, const gchar *subsystem)
 {
 	FuPluginPrivate *priv = GET_PRIVATE (self);
+	if (priv->udev_subsystems == NULL)
+		priv->udev_subsystems = g_ptr_array_new_with_free_func (g_free);
```

Comment:
I'm not convinced this will actually help anything.  the engine calls `fu_plugin_set_udev_subsystems` when plugins get loaded which will still initialize it.

Reply:
I think it's for the tests that just use FuPlugin and do not create a FuEngine.

Reply:
I don't think so?  I see the engine calling it when you do `fu_engine_load_plugins` which is part of `fu_engine_load`, should run in the main case

Reply:
Right, but in the self tests case we don't have an engine instance:

    #0  0x00007ffff7f74e5f in fu_plugin_add_udev_subsystem (self=0x43c160, subsystem=0x7ffff7f3c08f "drm") at ../libfwupdplugin/fu-plugin.c:1655
    #1  0x00007ffff7f370e5 in fu_plugin_init (plugin=0x43c160) at ../plugins/synaptics-mst/fu-plugin-synaptics-mst.c:172
    #2  0x00007ffff7f72b6e in fu_plugin_open (self=0x43c160, filename=0x43d050 "/home/hughsie/Code/fwupd/build/plugins/synaptics-mst/libfu_plugin_synaptics_mst.so", error=0x7fffffffd8a0)
        at ../libfwupdplugin/fu-plugin.c:448
    #3  0x0000000000403d22 in fu_plugin_synaptics_mst_none_func () at ../plugins/synaptics-mst/fu-self-test.c:67
    #4  0x00007ffff7c372be in g_test_run_suite_internal () at /lib64/libglib-2.0.so.0
    #5  0x00007ffff7c370bb in g_test_run_suite_internal () at /lib64/libglib-2.0.so.0
    #6  0x00007ffff7c370bb in g_test_run_suite_internal () at /lib64/libglib-2.0.so.0
    #7  0x00007ffff7c377aa in g_test_run_suite () at /lib64/libglib-2.0.so.0
    #8  0x00007ffff7c377c5 in g_test_run () at /lib64/libglib-2.0.so.0
    #9  0x0000000000404301 in main (argc=1, argv=0x7fffffffdd68) at ../plugins/synaptics-mst/fu-self-test.c:131